# https://github.com/coredns/coredns/pull/6384
From 81938bca2041c9dddc1171d56f609191313b8b9d Mon Sep 17 00:00:00 2001
From: Rahil Bhimjiani <rahil3108@gmail.com>
Date: Sun, 5 Nov 2023 21:53:05 +0530
Subject: [PATCH] Makefile: allow to use custom GITCOMMIT & BUILDOPTS

?= means If GITCOMMIT/BUILDOPTS is defined then use that otherwise fallback to default

Signed-off-by: Rahil Bhimjiani <rahil3108@gmail.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 43d03e207ba..10fca3da1a1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,9 @@
 # Makefile for building CoreDNS
-GITCOMMIT:=$(shell git describe --dirty --always)
+GITCOMMIT?=$(shell git describe --dirty --always)
 BINARY:=coredns
 SYSTEM:=
 CHECKS:=check
-BUILDOPTS:=-v
+BUILDOPTS?=-v
 GOPATH?=$(HOME)/go
 MAKEPWD:=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
 CGO_ENABLED?=0
